File: frontend/src/api/events.ts
import { useQuery } from "@tanstack/react-query";
import { apiFetch } from "./client";
import { useAgentStatus } from "./agent";

export interface AppEvent {
  id: string;
  event_type: string;
  level: "running" | "success" | "error" | "warning" | "info";
  title: string;
  detail: string | null;
  component: string | null;
  tenant: string | null;
  timestamp: string;
}

export function useEvents(limit = 100) {
  const { data: agentStatus } = useAgentStatus();
  return useQuery<AppEvent[]>({
    queryKey: ["events"],
    queryFn: () => apiFetch<AppEvent[]>(`/events?limit=${limit}`),
    refetchInterval: agentStatus?.is_running ? 3_000 : 10_000,
  });
}

File: backend/app/api/events.py
from fastapi import APIRouter, Query
from pydantic import BaseModel

from app.services import db

router = APIRouter(tags=["events"])


class AppEvent(BaseModel):
    id: str
    event_type: str
    level: str
    title: str
    detail: str | None = None
    component: str | None = None
    tenant: str | None = None
    timestamp: str


@router.get("/events", response_model=list[AppEvent])
def list_events(limit: int = Query(default=100, le=500)):
    return db.get_recent_events(limit)
